sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -427,22 +440,14 @@ public WritableByteChannel create(GcsPath path, String type) throws IOException
    */
   public WritableByteChannel create(GcsPath path, String type, Integer uploadBufferSizeBytes)
       throws IOException {
-    GoogleCloudStorageWriteChannel channel =
-        new GoogleCloudStorageWriteChannel(
-            executorService,
-            storageClient,
-            new ClientRequestHelper<>(),
-            path.getBucket(),
-            path.getObject(),
-            type,
-            /* kmsKeyName= */ null,
-            AsyncWriteChannelOptions.newBuilder().build(),
-            new ObjectWriteConditions(),
-            Collections.emptyMap());
+    WritableByteChannel channel = getCloudStorage().create(new StorageResourceId(path.getBucket()));
     if (uploadBufferSizeBytes != null) {
-      channel.setUploadBufferSize(uploadBufferSizeBytes);
+      if (channel instanceof GoogleCloudStorageWriteChannel) {
```

Comment:
I'm a bit hesitant to have this but i'm okay if we have to. @chamikaramj Do you think this is still necessary? If this is required, we might consider to have new API in `getCloudStorage().create` to accept `uploadBufferSizeBytes` to avoid code for each type of channels.

Reply:
This seems to be replaced with using `AsyncWriteChannelOptions.setUploadChunkSize` in `GoogleCloudStorageOptions.WriteChannelOptions`.

Reply:
done